lib/gssapi/mech/gss_inquire_cred.c

```diff
@@ -174,6 +176,17 @@ gss_inquire_cred(OM_uint32 *minor_status,
 		}
 	}
 
+	if (found && mechanisms) {
```

Comment:
Ah, so the answer is "no" to the above question.  So the above code can be removed.

Reply:
I got a bit confused as to which no/above you mean...

Reply:
> But now we'll never have a mech cred element for a GM_USE_MG_CRED mech, no?

Reply:
Ah, I see what you mean.

Reply:
Sorry – I got the comments about the two bits of code confused. Yes, we will never have a mech cred element for a `GM_USE_MG_CRED` mech so this should become an assert instead. (Hmm, unless we need to import exported creds from a previous release? What are the lifecycle guarantees of `gss_export_cred()`?)

The `found` check is in case the mech's `gss_inquire_cred()` failed. But we can unconditionally add `GM_USE_MG_CRED` mechs to the list if you prefer.

Reply:
Re: `gss_export_cred()` we should workaround that on `gss_import_cred()` if we need to.

Reply:
We probably don't need to do anything about `gss_import_cred()` as SPNEGO isn't going to implement `gm_import_cred()`. But... given I just argued the opposite above, I'll add a check for `GM_USE_MG_CRED`.

Reply:
Ah, OK, so `gss_import_cred()` should skip `GM_USE_MG_CRED` mechs rather than returning `GSS_S_BAD_MECH`. But – should we be storing `neg_mechs` in the exported cred so it round-trips?

Reply:
I mean, if we wanted to be clever, we *could* store `neg_mechs` in the exported cred as a SPNEGO credential with just a list of mechs. That would let us import old exported creds (but it wouldn't work the other way). Whether we need to address this at all depends on whether the exported credential format is stable between releases?

Reply:
If we need (or want) to support import/cred creds in both directions with old versions then, I guess we need to recursively embed the credential handle (filtered by `neg_mechs`).

Reply:
I guess export cred should save the neg mechs.

Reply:
> But – should we be storing neg_mechs in the exported cred so it round-trips?

Yes, that'd be nice.

Reply:
Though it's hardly critical.  Who uses export/import creds?

Reply:
And no, I don't think we want to be backwards-compatible for IPC tokens with multiple versions of Heimdal installed on the same host.

Reply:
OK, we will now encode neg mechs as the SPNEGO mech with a list of OIDs. This format is incompatible with previous versions of Heimdal.